File: 04-factorial/factorial.lua
-- Lua program to compute factorial
-- Uses recursion technique
-- Created by M.V.Harish Kumar on 18-06-2023

function Factorial(n)
    return (n <= 1) and 1 or (n * Factorial(n-1))
end

io.write("Enter value for `n`: ")
local n = tonumber(io.read("*n"))/1
local ans = Factorial(n)
print(string.format("%d! = %f", n, ans))

File: 04-factorial/factorial.pl
# Perl program to compute factorial
# Created by M.V.Harish Kumar on 09-01-2025
use v5.40;

sub factorial($n) {
	return $n <= 1 ? 1 : $n * factorial($n-1);
}

print "Enter value for `n`: ";
chomp(my $n = <STDIN>);
say "$n! = @{[factorial $n]}"